include/seqan/index/find2_index_approx.h

```diff
@@ -0,0 +1,593 @@
+
+            // NOTE (cpockrandt): this might not be optimal yet! we have more edges than in the theoretical model,
+            // since we go down an edge before we check whether it can even work out!
+            if (!std::is_same<TDistanceTag, EditDistance>::value && minErrorsLeftInBlock > 0 && charsLeft - 1 < minErrorsLeftInBlock - delta)
+            {
+                continue;
+            }
+
+            int32_t needleLeftIt2 = needleLeftIt - !goToRight;
+            uint32_t needleRightIt2 = needleRightIt + goToRight;
+
+            if (needleRightIt - needleLeftIt == s.blocklength[blockIndex])
+            {
+                // leave the possibility for one or multiple deletions! therefore, don't change direction, etc!
+                if (std::is_same<TDistanceTag, EditDistance>::value)
+                {
+                    _schemeSearchDeletion(delegate, iter, needle, needleLeftIt2, needleRightIt2, errors + delta, s, blockIndex, TDir());
+                }
+                else
+                {
+                    uint8_t blockIndex2 = std::min(blockIndex + 1, static_cast<uint8_t>(s.u.size()) - 1);
+                    bool goToRight2 = s.pi[blockIndex2] > s.pi[blockIndex2 - 1];
+                    if (goToRight2)
+                    {
+                        _schemeSearch(delegate, iter, needle, needleLeftIt2, needleRightIt2, errors + delta, s, blockIndex2, Rev(), TDistanceTag());
+                    }
+                    else
+                    {
+                        _schemeSearch(delegate, iter, needle, needleLeftIt2, needleRightIt2, errors + delta, s, blockIndex2, Fwd(), TDistanceTag());
+                    }
+                }
+            }
+            else
+            {
+                _schemeSearch(delegate, iter, needle, needleLeftIt2, needleRightIt2, errors + delta, s, blockIndex, TDir(), TDistanceTag());
+            }
+
+            // Deletion
+            if (std::is_same<TDistanceTag, EditDistance>::value)
+            {
+                _schemeSearch(delegate, iter, needle, needleLeftIt, needleRightIt, errors + 1, s, blockIndex, TDir(), TDistanceTag());
+            }
+        } while (goRight(iter, TDir()));
+    }
+}
+
+template <typename TDelegate, typename TText, typename TIndex, typename TIndexSpec, typename TText2, size_t nbrBlocks, typename TDir, typename TDistanceTag>
+inline void _schemeSearchExact(TDelegate & delegate,
+                               Iter<Index<TText, BidirectionalIndex<TIndex> >, VSTree<TopDown<TIndexSpec> > > iter,
+                               TText2 const & needle,
+                               uint32_t const needleLeftIt,
+                               uint32_t const needleRightIt,
+                               uint8_t const errors,
+                               Search<nbrBlocks> const & s,
+                               uint8_t const blockIndex,
+                               TDir /**/,
```

Comment:
You should at `const &` to the tags, to avoid meta-function voodoo errors if some other Metafunction is not `const` aware and/or the reference is captured somehow in the Type itself.